sdcm/cluster.py

```diff
@@ -5006,7 +5031,12 @@ def node_setup(self, node, verbose=False, **kwargs):  # pylint: disable=unused-a
             self.log.info("Don't install anything because bare loaders requested")
             return
 
+        node.create_node_certificate(node.ssl_conf_dir / 'test.crt', node.ssl_conf_dir / 'test.key')
```

Comment:
so, when using docker based loaders we send certs, why creating them for vm based loaders? Shouldn't they also be sent later when created?
Please add a test with param `use_prepared_loaders: true` so we are sure it works in this case as well.

Reply:
Added check for `use_prepared_loaders` in LoaderSet.node_setup, to put certificate to loader VM when this flag is enabled. The test for this is below (I added a temporary test config with this flag enabled to run with PR-provision test, as existing config with the flag are 5h+ long).
[PR-provision-test config + use_prepared_loaders + server_encrypt + client_encrypt + mtls](https://jenkins.scylladb.com/job/scylla-staging/job/dimakr/job/certs-rework-tests/24/)

Repeated a test where docker based loader is used:
[DecommissionMonkey nemesis + server_encrypt + client_encrypt](https://jenkins.scylladb.com/job/scylla-staging/job/dimakr/job/certs-rework-tests/25/)